client/tx/tx.go

```diff
@@ -335,43 +376,40 @@ func Sign(txf Factory, name string, tx client.TxBuilder) error {
 	}
 
 	pubKey := key.GetPubKey()
-	sigData := &signing.SingleSignatureData{
-		SignMode:  signMode,
-		Signature: nil,
-	}
-	sig := signing.SignatureV2{
-		PubKey: pubKey,
-		Data:   sigData,
-	}
-	err = tx.SetSignatures(sig)
```

Comment:
Please revert. Add tests for `SIGN_MODE_DIRECT` specifically if that's needed. (Just adding a review comment here to prevent this from getting merged like this.)